crates/ide_completion/src/render.rs

```diff
@@ -117,6 +117,17 @@ impl<'a> RenderContext<'a> {
         node.docs(self.db())
     }
 
+    fn expected_name(&self) -> Option<String> {
```

Comment:
I don't really like the duplication here with `expected_name_and_type`. I think ideally expected type would always just come from the field on `CompletionContext`. 

I spent some time trying to unify (or more accurately de-unify) `expected_name_and_type`, `expected_name`, and the `expected_type` field, but what I found was the type returned by `expected_name_and_type` is different in some cases from the type in the `expected_type` field, and various completions depend on those differences.

I think fully solving that problem is outside the scope of this PR. So perhaps this `expected_name` duplication is okay? I don't see an easy alternative that would be better, but I'm open to suggestions here.

Reply:
It's ok to duplicate for now, but we really should turn both into the fields. It's OK to change the results on existing tests, as long as the new result is still sensible.

Reply:
I've proposed a fix for this in #8008.